plugins/synology-ss/src/main.ts

```diff
@@ -0,0 +1,283 @@
+import sdk, { ScryptedDeviceBase, DeviceProvider, Settings, Setting, ScryptedDeviceType, VideoCamera, MediaObject, Device, ScryptedInterface, Camera, MediaStreamOptions, PictureOptions } from "@scrypted/sdk";
+import { recommendRebroadcast } from "../../rtsp/src/recommend";
+import {SynologyApiClient, SynologyCameraStream, SynologyCamera} from "./api/synology-api-client";
+
+const { log, deviceManager, mediaManager } = sdk;
+
+class SynologyCameraDevice extends ScryptedDeviceBase implements Camera, Settings, VideoCamera {
+    private provider: SynologySurveillanceStation;
+    private streams: SynologyCameraStream[];
+
+    constructor(provider: SynologySurveillanceStation, nativeId: string, camera: SynologyCamera) {
+        super(nativeId);
+        this.provider = provider;
+
+        this.streams = [
+            { ...camera.stream1, id: '1' },
+            { ...camera.stream2, id: '2' },
+            { ...camera.stream3, id: '3' },
+        ].filter(s => !!s.resolution);
```

Comment:
I don't understand what is going on here without seeing the JSON, but my guess is that Synology always exposes 3 streams, which may or may not actually be valid (ie, no resolution).

Reply:
stream1/2/3 could be empty, so this logic is simply the most concise way I found to both set id labels on them and put them all in an array.